### PR TITLE
Adding an initial unit tests writeup.

### PR DESCRIPTION
**Pull request by stirlaf**
Adding an initial unit tests writeup.
#6

### CODE REVIEW

File: week1-webclient_unittests.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
+# Codess in the Classroom - Week 1: The Web Client
+
+## Adding unit tests for your pet component.
+
+At this stage you should already have a pet component created in the src folder. It's valuable to now develop test for this component so that any future changes do not break functionality you have already developed.
```
Comment (pviotti) [inline week1-webclient_unittests.md]: I suggest mentioning that tests are important also to validate the correctness of the current implementation. And maybe that they can also be used to "drive" the implementation (TDD)

File: week1-webclient_unittests.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+        describe('PetCard', () => {
+            it('should render the given name', () => {
+                const { getByText } = render(<PetCard pet={petMock}/>);
+                expect(getByText('Fido')).toBeInTheDocument();
+            });
+        });
+
+    - The first parameter passed to `describe` should be the name of the component that is being tested.
+    - The first parameter passed to `it` should describe in natuaral language (e.g. English) what expectation you are trying to test.
```
Comment (pviotti) [inline week1-webclient_unittests.md]: typo: natuaral -> natural

File: week1-webclient_unittests.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    - The second parameters passed to `describe` and `it` are callbacks executed by the testing framework that should contain the actual testing code.
+    - The line `const { getByText } = render(<PetCard pet={petMock}/>);` renders the PetCard using the mock data defined in 6 above, and returns a function `getByText` that allows you to check whether certain text values exist in the rendered component.
+    - The line `expect(getByText('Fido')).toBeInTheDocument();` defines an expectation that the text `Fido` should be in the rendered component.
+
+8. There are many different ways in which the rendered component and it's parts can be checked for validity.
```
Comment (pviotti) [inline week1-webclient_unittests.md]: typo: it's -> its

File: week1-webclient_unittests.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+To simulate a failure, try changing the unit test described above, so that the text that is expected is `Spot` instead of `Fido`, e.g:
+
+    expect(getByText('Spot')).toBeInTheDocument();
+
+This will print out detailed information about the test that failed including the expectation that wasn't met.
+
+### Best practices when writing tests.
+
+The description that you give your tests should be similar to a specification. A good template to consider is `Code a should do action b in case of c`. The code in the test should of do checks that match the description. Since requirements tend to change over time or even get lost, unit tests can serve as an accurate and up to date specification of the expected behavior of the application. Since they are checked in with the code, they are easier to keep in sync with the code and maintain than purely textual specifications.
```
Comment (pviotti) [inline week1-webclient_unittests.md]: typo: should of do

File: week1-webclient_unittests.md [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+
+### Full sample solution.
```
Comment (pviotti) [inline week1-webclient_unittests.md:121]: Suggestion: enclose this into `<details><summary></summary><details>` tags as I did for https://github.com/codessintheclassroom/classroom-material/blob/master/week1-webclient_ui.md mostly for coherence of style

**Review by pviotti — APPROVED** [review]
LGTM :+1:

File: week1-webclient_unittests.md (hunk 2 above)
Comment (stirlaf) [reply week1-webclient_unittests.md]: Thanks, fixed.

File: week1-webclient_unittests.md (hunk 3 above)
Comment (stirlaf) [reply week1-webclient_unittests.md]: Thanks, fixed.

File: week1-webclient_unittests.md (hunk 4 above)
Comment (stirlaf) [reply week1-webclient_unittests.md]: Thanks, fixed.

File: week1-webclient_unittests.md (hunk 5 above)
Comment (stirlaf) [reply week1-webclient_unittests.md:121]: Oh yeah, that's nifty, updated to use it.

File: week1-webclient_unittests.md (hunk 1 above)
Comment (stirlaf) [reply week1-webclient_unittests.md]: Makes sense.  Added.

**Pull request merged**